cmd/containerd-shim-runhcs-v1/service.go

```diff
@@ -369,15 +371,11 @@ func (s *service) DiagStacks(ctx context.Context, req *shimdiag.StacksRequest) (
 	}
 	resp := &shimdiag.StacksResponse{Stacks: string(buf)}
 
-	if !s.isSandbox {
-		return resp, nil
-	}
-
-	sp, err := s.getPod()
-
-	if p, ok := sp.(*pod); ok {
-		resp.GuestStacks, err = p.host.DumpStacks(ctx)
+	t, _ := s.getTask(s.tid)
```

Comment:
What is the motivation for turning the dump stacks code this into a task function?

Reply:
So a `pod` is only for CRI but a `task` can be both containerd and CRI. So what this code does is really say "get me the owning task of the shim". That task may be the sandbox or it maybe be standalone. But we can support calling stacks on both at the same time. So no need to limit this to CRI